presto-orc/src/main/java/com/facebook/presto/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -73,6 +75,8 @@ public SliceDictionaryColumnWriter(
         this.dictionaryLengthStream = createLengthOutputStream(columnWriterOptions, dwrfEncryptor, orcEncoding);
         this.stringStatisticsLimitInBytes = toIntExact(columnWriterOptions.getStringStatisticsLimit().toBytes());
         this.statisticsBuilder = newStringStatisticsBuilder();
+        this.sortDictionaryKeys = columnWriterOptions.isStringDictionarySortingEnabled();
+        checkState(sortDictionaryKeys || orcEncoding == DWRF, "Disabling sort is only supported in DWRF format");
```

Comment:
We miss a negate in front of `sortDictionaryKeys`?

Reply:
There are only two conditions allowed.

1. sortDictionarykeys is true
2. sortDictionaryKeys is false, then orcEncoding must be DWRF.